.github/workflows/main.yml

```diff
@@ -35,6 +35,15 @@ jobs:
       - name: Run Tests on React Client
         run: npm run testcov
         working-directory: ./client
+  trigger-deploy-doc:
+    name: Generate and deploy docs for researchify.github.io
+    runs-on: ubuntu-latest
+    steps:
+      - uses: peter-evans/repository-dispatch@v1
+        with:
+          token: ${{ secrets.DOCS_WORKFLOW_PAT }}
+          repository: Researchify/researchify.github.io
+          event-type: trigger-deploy
```

Comment:
curious: are we going to be deploying docs on commits to non-`main` branches as well?
Also, I'm wondering how this CI job relates to the `script` defined for the docs? https://github.com/Researchify/Researchify/blob/be7004aaaa2de6f6040542d2c183ced1a4fb95dc/api/package.json#L35
More specifically, where are we generating and deploying the docs here?

Reply:
I see that the github pages docs repo also has a CI script: https://github.com/Researchify/researchify.github.io/blob/main/.github/workflows/main.yml
I understand now. Is it worth documenting the relationship between these two CI scripts somewhere to make it easier to understand?

Reply:
1. Good catch - I plan to add a condition to only trigger if its main branch before this is merged. I don't think the script / scripts are related.
2. Yup I think I'll document the relationship in the readme of the docs repo